packages/rum-core/src/domain/vital/vitalCollection.spec.ts

```diff
@@ -1,30 +1,46 @@
-import type { Duration } from '@datadog/browser-core'
+import type { Duration, Subscription } from '@datadog/browser-core'
+import { mockClock, type Clock } from '@datadog/browser-core/test'
 import { clocksNow } from '@datadog/browser-core'
-import type { TestSetupBuilder } from '../../../test'
-import { setup } from '../../../test'
-import type { RawRumVitalEvent } from '../../rawRumEvent.types'
+import type { RawRumEvent, RawRumVitalEvent } from '../../rawRumEvent.types'
 import { VitalType, RumEventType } from '../../rawRumEvent.types'
+import type { RawRumEventCollectedData } from '../lifeCycle'
+import { LifeCycle, LifeCycleEventType } from '../lifeCycle'
+import type { PageStateHistory } from '../contexts/pageStateHistory'
 import { createVitalInstance, startVitalCollection } from './vitalCollection'
 
+const pageStateHistory: PageStateHistory = {
+  findAll: () => undefined,
+  addPageState: () => {},
+  stop: () => {},
+  wasInPageStateAt: () => false,
+  wasInPageStateDuringPeriod: () => false,
+}
+
 describe('vitalCollection', () => {
-  let setupBuilder: TestSetupBuilder
+  const lifeCycle = new LifeCycle()
+  let rawRumEvents: Array<RawRumEventCollectedData<RawRumEvent>> = []
+  let clock: Clock
   let vitalCollection: ReturnType<typeof startVitalCollection>
   let wasInPageStateDuringPeriodSpy: jasmine.Spy<jasmine.Func>
+  let eventsSubscription: Subscription
 
   beforeEach(() => {
-    setupBuilder = setup()
-      .withFakeClock()
-      .beforeBuild(({ lifeCycle, pageStateHistory }) => {
-        wasInPageStateDuringPeriodSpy = spyOn(pageStateHistory, 'wasInPageStateDuringPeriod')
-        vitalCollection = startVitalCollection(lifeCycle, pageStateHistory)
-      })
+    clock = mockClock()
+    wasInPageStateDuringPeriodSpy = spyOn(pageStateHistory, 'wasInPageStateDuringPeriod')
+    vitalCollection = startVitalCollection(lifeCycle, pageStateHistory)
+    eventsSubscription = lifeCycle.subscribe(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, (data) => {
```

Comment:
**💬 suggestion:** ‏ ‏with this change, we are losing event format validation (https://github.com/DataDog/browser-sdk/blob/2d68a137b6a3f8ab248e97ceedd85a2f3fae9ee3/packages/rum-core/test/testSetupBuilder.ts#L134)

Maybe you could add it back in this test:

```ts
lifeCycle.subscribe(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, (data) => {
  rawRumEvents.push(data)
  validateRumEventFormat(data)
})
```

This is a bit inconvenient though, we could think of something else in the future.